### PR TITLE
WIP add upload to storj flow

### PR DESCRIPTION
**Pull request by tushar-saxena-yral**

### CODE REVIEW

**Review by psykomal — CHANGES_REQUESTED** [review]
small change suggestion. 

Also make sure the pipeline is running end2end in preview by uploading video in ssr
make sure these happen :- 
- video getting added to storj (I know you tested this, but is it also tested after secrets got added ?)
- video getting added to google cloud storage  (ask Saikat to add in our google project) - yral-videos bucket name
- video nsfw details getting inserted in nsfw table - table name :  hot-or-not-feed-intelligence.yral_ds.video_nsfw

File: src/consts.rs [hunk 1]
```diff
@@ -19,7 +19,8 @@ pub const CLOUDFLARE_ACCOUNT_ID: &str = "a209c523d2d9646cc56227dbe6ce3ede";
 
 pub const ICP_LEDGER_CANISTER_ID: &str = "ryjl3-tyaaa-aaaaa-aaaba-cai";
 
+// TODO: change this back to https://icp-off-chain-agent.fly.dev/ before merging
 pub static OFF_CHAIN_AGENT_URL: Lazy<Url> =
-    Lazy::new(|| Url::parse("https://icp-off-chain-agent.fly.dev/").unwrap());
+    Lazy::new(|| Url::parse("https://pr-102-yral-dapp-off-chain-agent.fly.dev/").unwrap());
```
[the remark this answers is not in the document]
Comment (tushar-saxena-yral) [reply src/consts.rs:24]: will revert back once at the very end, once i have tested with the creds that saikat provides

File: src/events/nsfw.rs [hunk 2]
```diff
@@ -177,30 +178,51 @@ struct VideoNSFWData {
...
 
-#[cfg(feature = "local-bin")]
-pub async fn nsfw_job(
-    State(state): State<Arc<AppState>>,
-    Json(payload): Json<VideoRequest>,
-) -> Result<Json<serde_json::Value>, AppError> {
-    Err(anyhow::anyhow!("not implemented for local binary").into())
-}
-
-#[cfg(not(feature = "local-bin"))]
```
[the remark this answers is not in the document]
Comment (tushar-saxena-yral) [reply src/events/nsfw.rs:188]: oh it is, i will add this back. removed it cuz it was messing with my lsp kek

File: .github/workflows/deploy-on-merge-to-main.yml [hunk 3]
```diff
@@ -70,6 +73,9 @@ jobs:
           CF_IMAGES_API_TOKEN: ${{ secrets.CLOUDFLARE_IMAGES_READ_AND_WRITE_API_TOKEN }}
           NSFW_GRPC_TOKEN: ${{ secrets.TOKEN_TO_SIGN_OUTGOING_CALLS_TO_NSFW_DETECT_SERVICE }}
           QSTASH_AUTH_TOKEN: ${{ secrets.QSTASH_TOKEN }}
+          STORJ_ACCESS_KEY_ID: ${{ secrets.STORJ_ACCESS_KEY_ID }}
+          STORJ_SECRET_KEY: ${{ secrets.STORJ_SECRET_KEY }}
+          STORJ_GATEWAY_URL: ${{ secrets.STORJ_GATEWAY_URL }}
```
[the remark this answers is not in the document]
Comment (tushar-saxena-yral) [reply .github/workflows/deploy-on-merge-to-main.yml:78]: makes sense, but how will they reach the fly machine? im sorry im not too familiar with secrets in github ci/cd

**Comment by tushar-saxena-yral** [comment]
> * video getting added to storj (I know you tested this, but is it also tested after secrets got added ?)

i have tested this with my creds added as secrets.

> * video getting added to google cloud storage

that's a really good concern, will confirm with saikat

File: .github/workflows/deploy-on-merge-to-main.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (komal-sai-yral) [reply .github/workflows/deploy-on-merge-to-main.yml:78]: depends on whether Saikat embedded them as secrets or env vars. once check with him 

if it is env var, use it like this https://github.com/yral-dapp/nsfw_detect/blob/0dd87313f80d86d0df797e874d3c608f2bd1e9c3/.github/workflows/deploy-on-merge-to-main.yml#L33

[the remark this answers is not in the document]
Comment (komal-sai-yral) [reply .github/workflows/deploy-on-merge-to-main.yml:78]: as to how they get added to fly env vars, its due to the cmd above these lines : flyctl secrets set

**Comment by tushar-saxena-yral** [comment]
@komal-sai-yral so i believe there are no more code changes required, only testing with real creds once and making sure:
- storj upload works
- gcp upload isn't broken
- nsfw detection isn't broken

@saikat-das-yral cc

**Comment by tushar-saxena-yral** [comment]
@saikat-das-yral
We need the following details for accessing the gateway. 
```bash
          STORJ_ACCESS_KEY_ID # can be an env var
          STORJ_SECRET_KEY # must be a secret
          STORJ_GATEWAY_URL # can be an env var
```

To self host the gateway you can follow the guide [here](https://storj.dev/learn/self-host/gateway-st).

Tldr:
- Create an account on storj
- Create an access grant
- Setup the gateway with storj's satellite (or host your own satellite https://storj.dev/learn/concepts/satellite)
- The access key and secret key will be outputted by the gateway program (https://storj.dev/learn/self-host/gateway-st#run-gateway-st)

## On hosting your own satellite
While it is possible to host your own satellite, it is a big can of worms. But here's some reference material:
- Announcement: https://forum.storj.io/t/announcing-community-satellite-pilot-program/21666
- https://forum.storj.io/t/how-to-run-a-satellite-node/26158/2
